packages/@uppy/thumbnail-generator/src/index.test.js

```diff
@@ -244,19 +249,33 @@ describe('uploader/ThumbnailGeneratorPlugin', () => {
     })
   })
 
-  describe('getProportionalHeight', () => {
-    it('should calculate the resized height based on the specified width of the image whilst keeping aspect ratio', () => {
-      const core = new MockCore()
-      const plugin = new ThumbnailGeneratorPlugin(core)
-      expect(
-        plugin.getProportionalHeight({ width: 200, height: 100 }, 50)
-      ).toEqual(25)
-      expect(
-        plugin.getProportionalHeight({ width: 66, height: 66 }, 33)
-      ).toEqual(33)
-      expect(
-        plugin.getProportionalHeight({ width: 201.2, height: 198.2 }, 47)
-      ).toEqual(46)
+  describe('getProportionalDimensions', () => {
+    const core = new MockCore()
+    const plugin = new ThumbnailGeneratorPlugin(core)
+
+    function resize (image, width, height) {
+      return plugin.getProportionalDimensions(image, width, height)
+    }
+
+    it('should calculate the thumbnail dimensions based on the width whilst keeping aspect ratio', () => {
+      expect(resize({ width: 200, height: 100 }, 50)).toEqual({ width: 50, height: 25 })
+      expect(resize({ width: 66, height: 66 }, 33)).toEqual({ width: 33, height: 33 })
+      expect(resize({ width: 201.2, height: 198.2 }, 47)).toEqual({ width: 47, height: 46 })
+    })
+
+    it('should calculate the thumbnail dimensions based on the height whilst keeping aspect ratio', () => {
+      expect(resize({ width: 200, height: 100 }, null, 50)).toEqual({ width: 100, height: 50 })
+      expect(resize({ width: 66, height: 66 }, null, 33)).toEqual({ width: 33, height: 33 })
+      expect(resize({ width: 201.2, height: 198.2 }, null, 47)).toEqual({ width: 48, height: 47 })
+    })
+
+    it('should calculate the thumbnail dimensions based on the default width if no custom width is given', () => {
+      plugin.defaultThumbnailDimension = 50
```

Comment:
I think we should use a separate instance here to avoid this mutation leaking into other tests, if we add more later. Right now the remaining tests don't depend on this value but who knows what will change in the future!